artifactory.py

```diff
@@ -1493,6 +1498,93 @@ class ArtifactoryOpensourceAccessor(_ArtifactoryAccessor):
     """
 
 
+# In Python 3.13, pathlib now reuses code from the glob package in order to implement
+# the Path.glob() method. There are two related classes in the glob package, _Globber
+# and _StringGlobber, where the former will delegate operations to the Path object while
+# the latter directly calls os.path functions, performing actual file system calls. The
+# private abstract base class of PurePath, PurePathBase, sets the _globber class
+# attribute to _Globber, while PurePath overrides it to be _StringGlobber.
+#
+# We create a custom subclass that explicitly subclasses _Globber and not
+# _StringGlobber, since we want the version that delegates file system operations to the
+# Path objects.
+#
+# In addition, we override _Globber.recursive_selector() with a copy of the original
+# code but with one modification. Inside the definition of the nested select_recursive()
+# function, we # add 1 to the original value of match_pos. The reason for this is that
+# the add_slash() method will not actually add a slash when the path object is an
+# instance of a Path subclass, since it will normally get normalized away. The match
+# position therefore needs to be incremented by 1 in order to account for the actual
+# slash character that appears when inspecting children of the current directory. This
+# isn't an issue in the actaul use of _Globber in Python, since it converts all paths to
+# strings, and the add_slash() will literally append a slash character to the string
+# path.
+class _ArtifactoryGlobber(glob._Globber if IS_PYTHON_3_13_OR_NEWER else object):
```

Comment:
If we inherit it from `object` always - will it act as usual?

Reply:
This class will be completely unused in the case of Python < 3.13. The conditional parent class is only to avoid hitting an import error in Python 3.12 and lower.